### DISCUSSION TITLE
Shared Teams File

### DISCUSSION THREAD

**Discussion by MikeTernosky** [dfinke/ImportExcel#1472] [Q&A]
Hi;
If I open and change a onedrive (teams) excel file that someone elsewhere also has open. After I make the changes and save and close when I come back the changes are reverted back to prior to my editing.
I can open the same excel file in excel manually with two people having access and the changes do save.

I have 
 - opened the file with importexcel module
 - edited /updated some lines on one worksheet
 - create a log entry on another worksheet
 - close the excelpackage
 - trigger the file opening (in actual excel at the end of the script), since getting the issue also try to hit save in excel at this point 
 - close the file
 - sometimes need to wait a few minutes but then upon reopening all my changes have reverted back to prior to running the script.

is there something I'm missing? do I need to open / close the file in a certain method for this?

Thanks
Mike

**Comment by dfinke** [comment]
please post a repro of the script and how you are working with the xlsx file.

**Reply by MikeTernosky** [reply]
posted the repo
- Read users from NEWHireRequest xlsx file
- open and find rows on PBX spreadsheet file Interviewers tab , that are marked as term or avail (to be edited for the newhires)
- update these lines (along with changing the DID ie. phone number, if necessary from newhires sheet (new hire needs a did get it from the extension file spare tab
- update the log tab in extension file
- close and create some one off csv files to be imported in a PBX (phone) system to change extensions, create users etc...

**Reply by MikeTernosky** [reply]
should add update the newhirerequest sheet with the extension from the extension file
this file has also been know to revert back when someone else has it open via onedrive.

**Reply by MikeTernosky** [reply]
believe the file NewHireRequest has "PBX Username" and DID field filled in - to run the script delete those two entries from NewHireRequest and the script should run and add to it (if there is a AVAIL or TERM extension in PBX Field Extension file.

**Comment by dfinke** [comment]
I won't be able to debug your process. I was wondering had a simple example of what you are doing in terms of opening and interacting with the xlsx. I could maybe see if that was an issue.

I don't work with xlsx files on onedrive or with teams.

**Comment by MikeTernosky** [comment]
Well
Stripped down

Open a file
Make some changes
Close the file

The files are located in a linked teams files folder – which is basically done through onedrive
Ie  so they ae basically mapped
***@***.***

And referenced as such
$Extfile = "$HOME\******\Field Technology - PBX Solution\BBD PBX Field extensions.xlsx"
$NewUsersfile = "$HOME\*****\Field Technology - User Solution\NewHireRequest - Test.xlsx"


The code is basically doing the below.
$Users = Import-Excel -Path $NewUsersfile -WorksheetName $NewUserWS           #Get List of New Users  <--- OPEN NEW USERS FILE
$PBXUsers = GetPBXNewUsers $Users                                             #Get list of New Uses requiring PBX <-- GET LIST OF NEW USERS
$Extensions = Open-ExcelPackage -Path $ExtFile -KillExcel                     #Open PBX File  <-- Open EXTENSIONS FILE
$Exts = $Extensions.Workbook.Worksheets[$ExtIntsWS]
$AvailExts = GetAvailableExt $Exts                                            #<-- get list of available extension from main list of all ext
UpdateUserRows $PBXUsers $Extensions                                          #<-- make certain rows in extension file as used and add details to those rows
UpdateLogs $PBXUsers $Extensions                                              #<-- add detail log message to logs worksheet in ext file
Close-ExcelPackage $Extensions -Show                                          #<-- close Extension file
UpdateNewUserRequest $PBXUsers $NewUsersfile                                  #<-- Update new user request file with new extensions found/edited above

if someone else has the file open (can see their pic and as teams/onedrive files goes , many people can have the file open and make changes at the same time), the changes do get made and I can open the file and see the changes
but in a few minutes the changes get wiped out and the file(s) revert to their original format.

It would appear the file is not getting opened in shared mode as I presume (not a onedrive expert myself) OneDrive / teams does. Maybe you know of something like opening the link directly to sharepoint would work
I am hoping there is a way to do this with importexcel, otherwise unfortunately in todays world with all the file sharing/editing going on at once – I will need to find a new method.

Really like importexcel but the shared file opening / saving is just too big if there is any other option, which I have yet to find as well though… continuing to try to work this out.

Here’s hoping you can see something – have heard of this type of thing occurring or some suggestion.
